File: app/src/main/java/com/datarockets/mnchkn/di/AppModule.java
package com.datarockets.mnchkn.di;

import com.datarockets.mnchkn.MunchkinApplication;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {

    private MunchkinApplication mApplication;

    public AppModule(MunchkinApplication application) {
        this.mApplication = application;
    }

    @Provides
    @Singleton
    public MunchkinApplication providesApplication() {
        return mApplication;
    }

}

File: app/src/main/kotlin/com/datarockets/mnchkn/di/AppModule.kt
package com.datarockets.mnchkn.di

import android.content.Context
import com.datarockets.mnchkn.MunchkinApplication
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module class AppModule(private val mApplication: MunchkinApplication) {

    @Provides @Singleton
    fun providesApplication(): MunchkinApplication {
        return mApplication
    }

    @Provides @Singleton
    fun providesContext(): Context {
        return mApplication.applicationContext
    }

}
